test/skimNtuple2018_HHbtag.cpp

```diff
@@ -4273,168 +4276,71 @@ int main (int argc, char** argv)
-                  cout<<"ECON MET"<<endl;
-                  cout<<"ECON (E,Px,Py,Pz,M) "<<","<<ptmiss.Px()<<","<<ptmiss.Py()<<endl;
-                  cout<<"ECON METCOV "<<endl;
-                  cout<<"ECON "<<metcov (0,0)<<"  "<<metcov (0,1)<<endl;// = theBigTree.MET_cov00->at (chosenTauPair) ;
-                  cout<<"ECON "<<metcov (1,0)<<"  "<<metcov (1,1)<<endl;// = theBigTree.MET_cov10->at (chosenTauPair) ;
-                  cout<<"ECON tau1, tau2, b1, b2"<<endl;
-                  cout<<"ECON ";
-                  tlv_firstLepton.Print();
-                  cout<<"ECON ";
-                  tlv_secondLepton.Print();
-                  cout<<"ECON ";
-                  tlv_firstBjet.Print();
-                  cout<<"ECON ";
-                  tlv_secondBjet.Print();
-                  wrongHHK=true;
-                }
-              if(!wrongHHK)
-                {
-                  HHKmass = kinFits.getMH () ;
-                  HHKChi2 = kinFits.getChi2 () ;
-                }
-              else
-                {
-                  if(isOS)HHKmass = -333;
-                }
+          {
+            if(DEBUG)
+              {
+                 cout<<"---Kinfit Input debug---"<<endl;
+                 cout<<"L1"<<endl;
+                 cout<<"(pt,eta,phi,e) "<<tlv_firstLepton.Pt()<<","<<tlv_firstLepton.Eta()<<","<<tlv_firstLepton.Phi()<<","<<tlv_firstLepton.E()<<endl;
+                 cout<<"L2"<<endl;
+                 cout<<"(pt,eta,phi,e) "<<tlv_secondLepton.Pt()<<","<<tlv_secondLepton.Eta()<<","<<tlv_secondLepton.Phi()<<","<<tlv_secondLepton.E()<<endl;
+                 cout<<"B1"<<endl;
+                 cout<<"(pt,eta,phi,e,res) "<<tlv_firstBjet.Pt()<<","<<tlv_firstBjet.Eta()<<","<<tlv_firstBjet.Phi()<<","<<tlv_firstBjet.E()<<","<<bjet1_JER<<endl;
+                 cout<<"B2"<<endl;
+                 cout<<"(pt,eta,phi,e,res) "<<tlv_secondBjet.Pt()<<","<<tlv_secondBjet.Eta()<<","<<tlv_secondBjet.Phi()<<","<<tlv_secondBjet.E()<<","<<bjet2_JER<<endl;
+                 cout<<"MET"<<endl;
+                 cout<<"(Px,Py) "<<","<<ptmiss.Px()<<","<<ptmiss.Py()<<endl;
+                 cout<<"METCOV "<<endl;
+                 cout<<metcov(0,0)<<"  "<<metcov(0,1)<<endl;
+                 cout<<metcov(1,0)<<"  "<<metcov(1,1)<<endl;
+               }
+
+            KinFitter fitter(tlv_firstBjet, tlv_secondBjet, tlv_firstLepton, tlv_secondLepton, ptmiss, stableMetCov, bjet1_JER, bjet2_JER);
+            KinFitter fitterRaw(tlv_firstBjet, tlv_secondBjet, tlv_firstLepton, tlv_secondLepton, ptmiss, stableMetCov, bjet1_JER, bjet2_JER);
+
+            wrongHHK = fitter.fit(whichSgnHp);
+            if(!wrongHHK) {
+              HHKmass = fitter.getMH();
+              HHKChi2 = fitter.getChi2();
+            }
+            else {
+              if(isOS)HHKmass = -333;
```

Comment:
If I remember correctly the message from Giles, in the case of non-convergence, the mass should be set to zero, this is what the new DNN trainings assume.

Reply:
Ok, will double-check and fix